src/cljs/rems/table.cljs

```diff
@@ -58,205 +21,276 @@
-                           :description "bar"}))
-      "doesn't match with value if filter-value is provided")
-  (is (not (match-filter? {:id {:value :id}
-                           :description {:value :description
-                                         :filterable? false}}
-                          "shouldnotmatch"
-                          {:id "foo"
-                           :description "shouldnotmatch"}))
-      "doesn't match if filterable is false"))
+(rf/reg-sub
+ ::sorting
+ (fn [db [_ table]]
+   (or (get-in db [::sorting (:id table)])
+       {:sort-order (or (:default-sort-order table)
+                        :asc)
+        :sort-column (or (:default-sort-column table)
+                         (-> table :columns first :key))})))
 
-(defn match-filters? [column-definitions filters item]
-  (every? (fn [filter-word]
-            (match-filter? column-definitions filter-word item))
-          (str/split filters #"\s+")))
+(rf/reg-event-db
+ ::set-filtering
+ (fn [db [_ table filtering]]
+   (assoc-in db [::filtering (:id table)] filtering)))
 
-(defn apply-filtering [column-definitions filters items]
-  (filter #(match-filters? column-definitions filters %) items))
+(rf/reg-sub
+ ::filtering
+ (fn [db [_ table]]
+   (get-in db [::filtering (:id table)])))
 
-(s/defschema Applicable
-  (s/cond-pre s/Keyword (s/pred fn?)))
+(defn- default-if-missing [m k make-default]
+  (if (contains? m k)
+    m
+    (assoc m k (make-default m))))
 
-(s/defschema TableParams
-  {:column-definitions {s/Keyword {(s/optional-key :header) s/Any
-                                   (s/optional-key :value) Applicable
-                                   (s/optional-key :values) Applicable
-                                   (s/optional-key :sortable?) s/Bool
-                                   (s/optional-key :sort-value) Applicable
-                                   (s/optional-key :filterable?) s/Bool
-                                   (s/optional-key :filter-value) Applicable
-                                   (s/optional-key :class) (s/cond-pre s/Str Applicable)}}
-   :visible-columns [s/Keyword]
-   (s/optional-key :sorting) {(s/optional-key :initial-sort) [{:sort-column s/Keyword
-                                                               (s/optional-key :sort-order) (s/enum :asc :desc)}]
-                              (s/optional-key :sort-column) s/Keyword
-                              (s/optional-key :sort-order) (s/enum :asc :desc)
-                              (s/optional-key :set-sorting) Applicable}
-   (s/optional-key :filtering) {(s/optional-key :filters) s/Str
-                                (s/optional-key :filters-new) s/Str
-                                (s/optional-key :show-filters) s/Bool
-                                (s/optional-key :set-filtering) Applicable}
-   :id-function Applicable
-   :items [s/Any]
-   (s/optional-key :class) s/Str})
+(defn- apply-row-defaults [row]
```

Comment:
This seems very complicated and I don't see a clear need why there is a custom `default-if-missing`.